2009-flu-uk.qmd

```diff
@@ -1 +1,437 @@
+
+```{r, echo = TRUE}
+seir <- odin2::odin({
+# initial conditions
+initial(S) <- (1 - 2 * alpha) * N
+initial(E) <- alpha * N
+initial(I) <- alpha * N
+initial(R) <- 0
+initial(incidence, zero_every = 7) <- 0
+
+# equations
+deriv(S) <- - hol * beta * S * I / N
+deriv(E) <- hol * beta * S * I / N - gamma * E
+deriv(I) <- gamma * E - sigma * I
+deriv(R) <- sigma * I
+deriv(incidence) <- gamma * E
+
+# parameter values
+R_0 <- parameter(1.5)
+L <- 1
+D <- 1.25
+alpha <- parameter(1e-4) # initial proportion
+N <- 55000000
+
+# convert parameters
+hol <- interpolate(h_times, h_values, "constant")
+h_times <- parameter()
+h_values <- parameter()
+dim(h_times) <- parameter(rank=1)
+dim(h_values) <- length(h_times)
+gamma <- 1 / L
+sigma <- 1 / D
+beta <- R_0 * sigma
+
+# observation model
+rho <- parameter(0.1)
+eta <- parameter(10)
+Cases <- data()
+Cases ~ NegativeBinomial(size = eta, mu = rho*incidence)
+})
+```
+
+### Running the model
+
+Once the model is compiled, it is possible to generate one (or more!) instance of your model by using the `dust_system_create()` and your model generator.
+
+```{r}
+mod <- dust_system_create(seir,
+                          pars = list(h_times = hol_t, h_values = hol_v))
+```
+
+```{r}
+# A vector with parameter values
+# First parameter -> proportion of population initially infected
+# Second argument -> R_0
+# Third argument -> Proportion of infection detected 
+par0 <- c(5e-5, 1.3, 0.1, 10)
+
+# This vector should be passed as a list
+dust_system_update_pars(sys = mod,
+                        pars = list(alpha=par0[1],
+                                    R_0=par0[2],
+                                    h_times = hol_t,
+                                    h_values= hol_v))
+```
+
+Then the model can be run by specifying the time points between which to integrate the ODEs. Note that the first point (here 0) set the inital time of the integration. Once the time vector t is defined we can run the model by using the "$run(t)" method.
+```{r}
+t <- c(0,data$Days)
+dust_system_set_state_initial(mod)
+y <- dust_system_simulate(mod, t)
+```
+
+Let's plot the I compartment over time.
+
+```{r}
+plot(t, dust_unpack_state(mod, y)$incidence, type = "l", col = "red", lwd = 2)
+```
+
+It looks a bit like the actual epidemic; a good sign, but let's compare our model with the data.
+
+```{r}
+plot(t, dust_unpack_state(mod, y)$incidence*par0[3], type = "l", col = "red", lwd = 2)
+points(data$Days, data$Cases)
+```
+
+We seem to be overpredicted the expected number of cases by a factor two or so, but the MCMC pipeline should allow us to infer the distribution of the parameters consistent with the observations.
```

Comment:
"be overestimating" or "have overestimated"

Reply:
Corrected